samcli/lib/cookiecutter/interactive_flow_creator.py

```diff
@@ -0,0 +1,80 @@
+        in the file, i.e. the default next-question of a given question will be the next one defined in the file,
+        while also respecting the question-defined _next_question map if provided.
+
+        Parameters:
+        ----------
+        flow_definition_path: str
+            A path to a json/yaml file that defines the questions of the flow. the file is expected to be in the
+            following format:
+            {
+                "Questions": [
+                    {
+                      "key": "key of the corresponding cookiecutter config",
+                      "question": "the question to prompt to the user",
+                      "kind": "the kind of the question, for example: confirm",
+                      "isRequired": true/false,
+                      # optional branching logic to jump to a particular question based on the answer. if not given
+                      # will automatically go to next question
+                      "nextQuestion": {
+                        "True": "key of the question to jump to if the user answered 'Yes'",
+                        "False": "key of the question to jump to if the user answered 'Yes'",
+                      }
+                    },
+                    ...
+                ]
+            }
+
+        Returns: InteractiveFlow(questions={k1: q1, k2: q2, ...}, first_question_key="first question's key")
+        """
+        questions, first_question_key = InteractiveFlowCreator._load_questions(flow_definition_path)
+        return InteractiveFlow(questions=questions, first_question_key=first_question_key)
+
+    @staticmethod
+    def _load_questions(flow_definition_path: str) -> Tuple[Dict[str, Question], str]:
+        previous_question: Optional[Question] = None
+        first_question_key: str = ""
+        questions: Dict[str, Question] = {}
+        template_data = get_template_data(flow_definition_path)
+
+        for question in template_data.get("Questions"):
+            q = InteractiveFlowCreator._create_question_from_json(question)
+            if not first_question_key:
+                first_question_key = q.key
+            elif previous_question and not previous_question.default_next_question_key:
+                previous_question.set_default_next_question_key(q.key)
+            questions[q.key] = q
+            previous_question = q
+        return questions, first_question_key
+
+    @staticmethod
+    def _create_question_from_json(question_json: Dict) -> Question:
+        key = question_json["key"]
+        text = question_json["question"]
+        options = question_json.get("options")
+        default = question_json.get("default")
+        is_required = question_json.get("isRequired")
+        next_question_map = question_json.get("nextQuestion")
+        kind_str = question_json.get("kind")
+        kind = QuestionKind[kind_str] if kind_str else None
```

Comment:
nit: I think here you want to do `QuestionKind(kind_str)` instead of `[kind_str]`. The former is finding the enum by value, the latter is by name. I feel by value makes more sense. Since your enum's name and value are the same, it does not matter too much here.